### PR TITLE
Add Discover Page and implement API 

### PR DESCRIPTION
**Pull request by ysilksa**
- Added left side of Discover page. 
- Added DiscoverAPI.js to implement the API given for HW 5
- Plan to implement other user data into the Profile page, but for now, focusing on finishing the Discover page.

### CODE REVIEW

File: my-app/src/pages/Discover.js [hunk 1]
```diff
@@ -1,12 +1,144 @@
+import React, { useState, useEffect } from 'react';
 import '../App.css';
 import './Discover.css'
+import { getAllUsers } from './DiscoverAPI.js'
```
Comment (aanandp123) [inline my-app/src/pages/Discover.js:4]: Doesnt seem like there's any routing being implemented using `react-router-dom`. I would suggest tying the pages together with this

File: my-app/src/pages/DiscoverAPI.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+// getAllUsers
+export async function getAllUsers() {
```
Comment (aanandp123) [inline my-app/src/pages/DiscoverAPI.js:2]: great way of separating out code into reusable clean functions!

File: my-app/src/pages/DiscoverAPI.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    const users = await fetch('https://disc-assignment-5-users-api.onrender.com/api/users', {
+      method: 'GET',
+    });
+    if (!users.ok) {
+      throw new Error('Failed to fetch users');
```
Comment (aanandp123) [inline my-app/src/pages/DiscoverAPI.js:7]: using` try-catch ` blocks instead  of this if statement handles runtime errors better

File: my-app/src/pages/Discover.js [hunk 4]
```diff
@@ -1,12 +1,144 @@
...
     </div>
     )
+}
+
+//
+// external functions
+//
+
+// heart icon for the post 
+function HeartIcon({size = 20, color = '#2A2B2A'}) {
```
Comment (aanandp123) [inline my-app/src/pages/Discover.js:129]: love this idea of using different intuitive icons!

**Comment by amyzliao** [comment]
You are free to merge this branch into main. We recommend you do so immediately, before you make any changes to your codebase for this assignment or future assignments.

If you have NOT started on assignment 6, the following steps will help you avoid tricky merge conflicts:
- If you have made local changes to your `hw-5` branch that are not published to github, first commit those changes and publish them to github. That should update this PR with your latest changes.
1. Merge this PR into main
2. Switch to the main branch: `git checkout main`
3. Download the latest main which should now have all your `hw-5` commits: `git pull`
4. Delete your local `hw-5` branch: `git branch -d hw-5`

If you HAVE started on assignment 6, the following steps will help you avoid tricky merge conflicts:
- **DO NOT MAKE ANY CHANGES TO YOUR `hw-5` BRANCH**. If you have already made changes to `hw-5`, you will have to resolve merge conflicts at step 6.
1. Merge this PR into main
2. Commit all your current changes for assignment 6 to the `hw-6` branch. Afterwards, make sure when you run `git status`, it says "nothing to commit, working tree clean".
3. Switch back to the main branch: `git checkout main`
4. Download the latest main which should now have all your `hw-5` commits: `git pull`
5. Switch back to your `hw-6` branch: `git checkout hw-6`
6. Rebase off of main: `git rebase main`
    - This is the trickiest bit. Ideally there should not be any conflicts.
    - If you have conflicts, the terminal output will tell you exactly which files conflict. Go to those files and resolve the conflicts. When you're done, run `git add . && git rebase --continue`
    - Either the rebase will finish or it will give you more merge conflicts to resolve. If it gives you more conflicts to resolve, repeat the above bullet point. It might make you resolve the same conflict more than once; that is normal behavior.
    - After you are fully finished with the rebase, run your code again and verify that it has the expected behavior. Bugs may have emerged during the rebase.
7. Force push your changes to the `hw-6` branch to github:
    - If this branch does not yet exist on github: `git push -u origin hw-6`
    - If this branch already exists on github: `git push --force-with-lease`

If you run into any issues, DM amy, ethan, or aanand on discord.

**Pull request merged**